agent_app_sample_code/utils/file_loading.py

```diff
@@ -82,21 +67,23 @@ def get_parser_udf(
     )
     return parser_udf
 
+def load_files_to_df(
+    spark: SparkSession,
+    source_path: str) -> DataFrame:
+    """
+    Load files from a directory into a Spark DataFrame.
+    Each row in the DataFrame will contain the path, length, and content of the file; for more
+    details, see https://spark.apache.org/docs/latest/sql-data-sources-binaryFile.html
+    """
 
-def load_uc_volume_to_delta_table(
```

Comment:
This util is hard to test because it does so much, it's easier to break this down into dataframe operations (load files into DF, apply parsing, etc) and test those